tests/GcodeLayerThreaderTest.cpp

```diff
@@ -0,0 +1,123 @@
+//Copyright (c) 2015 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "GcodeLayerThreaderTest.h"
+
+#include <iomanip>
+#include <sstream> // ostringstream
+#include <list>
+
+#include <../src/utils/intpoint.h>
+#include <../src/GcodeLayerThreader.h>
+
+
+
+namespace cura
+{
+    CPPUNIT_TEST_SUITE_REGISTRATION(GcodeLayerThreaderTest);
+
+void GcodeLayerThreaderTest::setUp()
+{
+    //Do nothing.
+}
+
+void GcodeLayerThreaderTest::tearDown()
+{
+    //Do nothing.
+}
+
+void GcodeLayerThreaderTest::test1()
+{
+    test(1000, 100, 10);
+}
+void GcodeLayerThreaderTest::test2()
+{
+    test(100, 1000, 10);
+}
+void GcodeLayerThreaderTest::test3()
+{
+    test(1000, 20, 1);
+}
+void GcodeLayerThreaderTest::test4()
+{
+    test(10000, 1000, 100);
+}
+
+
+
+void GcodeLayerThreaderTest::test(int avg_computation_time, int layer_count, int max_task_count)
+{
+//     srand(1);
```

Comment:
Ehh are you sure?
You were telling me that the test case shouldn't be random, so shouldn't I uncomment it then?